processors/processor.go

```diff
@@ -0,0 +1,37 @@
+package processors
+
+import (
+	"context"
+	"github.com/algorand/go-algorand/ledger/ledgercore"
+	"github.com/algorand/go-algorand/rpcs"
+	"github.com/algorand/indexer/data"
+	"github.com/algorand/indexer/plugins"
+)
+
+// BlockProcessor is the block processors interface
+type BlockProcessor interface {
```

Comment:
I'm confused, why are we using a different interface for the block processor?

Reply:
The BlockProcessor was the original interface made by Shiqi, the new "Processor" interface is the plugin interface.  The `blockProcessor` is the implementation of both BlockProcessor and Processor

Reply:
I need to review this more closely, but now that you've separated the Process/OnComplete functions I wonder if this interface could be part of the test code instead of the processor. You could return this from `setupIdb`

i.e.
```
func handle(proc processors.Processor, handler func(block *ledgercore.ValidatedBlock)) func(cert *rpcs.EncodedBlockCert) error {
    return func(cert *rpcs.EncodedBlockCert) error {
        blockData := proc.Process(BlockData{ Certificate: cert.cert, Block: cert.Block })
        handler(MakeValidatedBlock(blockData.Block, blockData.StateDelta))
        proc.OnComplete(blockData)
    }
}
```